src/node_mem.h

```diff
@@ -0,0 +1,41 @@
+#ifndef SRC_NODE_MEM_H_
+#define SRC_NODE_MEM_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include <stddef.h>
+
+namespace node {
+namespace mem {
+
+// Both ngtcp2 and nghttp2 allow custom allocators that
+// follow exactly the same structure and behavior, but
+// use different struct names. To allow for code re-use,
+// the NgLibMemoryManager template class can be used for both.
+
+template <typename Class, typename AllocatorStructName>
+class NgLibMemoryManager {
+ public:
+  // Class needs to provide these methods:
+  // void CheckAllocatedSize(size_t previous_size) const;
+  // void IncreaseAllocatedSize(size_t size);
+  // void DecreaseAllocatedSize(size_t size);
```

Comment:
Nit: perhaps declare these as pure virtual methods IIUC? (and maybe even `env()` but that would require a different name to avoid inheritance issues)

Reply:
They’re supposed to be called as inline methods, though, and virtual methods would kind of defeat the point of that?

Reply:
Oh, ok then.